src/test/java/ca/uhn/fhir/jpa/starter/OperationCollectDataTest.java

```diff
@@ -0,0 +1,200 @@
+    "spring.main.allow-bean-definition-overriding=true" })
+public class OperationCollectDataTest {
+
+  private static final org.slf4j.Logger ourLog = org.slf4j.LoggerFactory.getLogger(ExampleServerDstu2IT.class);
+  private IGenericClient ourClient;
+  private FhirContext ourCtx;
+  protected static CloseableHttpClient ourHttpClient;
+  protected static String ourServerBase;
+
+  private static String OBS_FILE_PATH = "src/test/resources/ObsBundle.json";
+
+  private static String PATIENT_FILE_PATH = "src/test/resources/PatientBundle.json";
+
+  private static String MEASURE_FILE_PATH = "src/test/resources/FhirMeasure.json";
+
+  private static String MEASURE_RESOURCE_ID = "TX-PVLS";
+
+  @LocalServerPort
+  private int port;
+
+  @Test
+  public void testCollectDataOperation() throws ClientProtocolException, IOException {
+    String paramName1 = "measureReport";
+    String paramName2 = "resource";
+    // Post the Measure Resource
+    Measure measure = readMeasureFromFile();
+    ourClient.update().resource(measure).withId(MEASURE_RESOURCE_ID).encodedJson().execute();
+    // post theobs BUndle
+    postResource(ourServerBase, OBS_FILE_PATH);
+    // fetch parameter reuslt from the Opration
+    Parameters result = fetchParameter(ourServerBase + "/Measure/" + MEASURE_RESOURCE_ID
+        + "/$collect-data?periodStart=2021-01-01&periodEnd=2021-01-31");
+
+    assertTrue(result.hasParameter(paramName1));
+    assertTrue(result.hasParameter(paramName2));
+    assertEquals(2, result.getParameter().size());
+
+    assertTrue(result.getParameter().get(0).getResource() instanceof MeasureReport);
+
+    assertTrue(result.getParameter().get(1).getResource() instanceof Bundle);
+    //get measure report from the Parameter Result
+    MeasureReport report = (MeasureReport) result.getParameter().get(0).getResource();
+    assertEquals(report.getEvaluatedResource().size(), 4);
+    assertEquals(report.getEvaluatedResource().size(), 4);
+    assertEquals(report.getMeasure(), "Measure/TX_PVLS");
+    assertEquals(report.getStatus(), MeasureReport.MeasureReportStatus.COMPLETE);
+    assertEquals(report.getType(), MeasureReport.MeasureReportType.DATACOLLECTION);
+
+    //get Observation Bundle from the Parameter Result
+    Bundle bundle = (Bundle)result.getParameter().get(1).getResource();
+    assertEquals(bundle.getEntry().size() , 2);
+    assertTrue(bundle.getEntry().get(0).getResource() instanceof Observation);
+    assertTrue(bundle.getEntry().get(1).getResource() instanceof Observation);
+  }
+
+  @BeforeEach
+  void beforeEach() {
+    ourServerBase = "http://localhost:" + port + "/fhir";
+    setHapiClient();
+    ourHttpClient = HttpClientBuilder.create().build();
+  }
+
+  private Parameters fetchParameter(String theUrl) throws IOException, ClientProtocolException {
+    Parameters parameters;
+    HttpGet get = new HttpGet(theUrl);
+
+    String username = "hapi";
+    String password = "hapi123";
```

Comment:
use these from properties

Reply:
@k-joseph ideally for now we use [hard coded ](https://github.com/openmrs/openmrs-contrib-plm-fhir-server/blob/master/src/main/java/ca/uhn/fhir/jpa/starter/AuthenticationInterceptor.java#L47  )credentials for Basic Auth just for the POC

we shall probably implement a better Authenication mechanism as a separate Task

Reply:
check my feedback on the above similar review